### PR TITLE
Attempt to migrate to Circle CI v2.0

### PR DESCRIPTION
**Pull request by davehunt**

### CODE REVIEW

**Comment by davehunt** [comment]
I saw @ckolos had a branch in progress, however it appears to be out of date. I'm not entirely clear on how we deploy Servicebook from Docker Hub, but I think this configuration would replicate the existing workflow.

**Comment by davehunt** [comment]
> Ah, I see the curl command is done differently. That explains that.

Yes, the job and remote docker run in separate evironments, so they can't see each other. See https://circleci.com/docs/2.0/building-docker-images/#separation-of-environments.

**Comment by ckolos** [comment]
+1

File: Pipfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [[source]]
-url = "https://pypi.org/simple"
+url = "https://pypi.python.org/simple"
```
Comment (davehunt) [inline Pipfile]: This was necessary as the current Pipfile hash in Pipfile.lock does not match. We can update this later.

**Comment by davehunt** [comment]
@ckolos same issue here with "Error: Cannot perform an interactive login from a non TTY device".

**Comment by ckolos** [comment]
Confirmed the fixes applied allowed a new container to be pushed to dockerhub

**Pull request merged**